docs/HowToGuides/GettingStarted.md

```diff
@@ -0,0 +1,609 @@
+space and (2) you need to maintain the two builds in sync, which needs extra
+care when moving across branches.
+
+### Troubleshooting build issues
+
+- Double-check that all projects are checked out at the right branches.
+  A common failure mode is using `git checkout` to change the branch only
+  for `swift` (often to a release branch), leading to an unsupported
+  configuration. See Step 4 of [Cloning the Project](#cloning-the-project)
+  on how to fix this.
+- Double-check that all your dependencies
+  [meet the minimum required versions](#spot-check-dependencies).
+- Check if there are spaces in the paths being used by `build-script` in
+  the log. While `build-script` should work with paths containing spaces,
+  sometimes bugs do slip through, such as
+  [SR-13441](https://bugs.swift.org/browse/SR-13441).
+  If this is the case, please [file a bug report][Swift JIRA] and change the path
+  to work around it.
+- Check that your `build-script` invocation doesn't have typos. You can compare
+  the flags you passed against the supported flags listed by
+  `utils/build-script --help`.
+- Check the error logs and see if there is something you can fix.
+  In many situations, there are several errors, so scrolling further back
+  and looking at the first error may be more helpful than simply looking
+  at the last error.
+- Check if others have encountered the same issue on the Swift forums or on
+  [Swift JIRA][].
+- Create a new Swift forums thread in the Development category. Include
+  information about your configuration and the errors you are seeing.
+  - You can [create a gist](https://gist.github.com) with the entire build
+    output and link it, while highlighting the most important part of the
+    build log in the post.
+  - Include the output of `utils/update-checkout --dump-hashes`.
+
+[Swift JIRA]: https://bugs.swift.org
+
+## Editing code
+
+### Setting up your fork
+
+If you are building the toolchain for development and submitting patches,
+you will need to setup a GitHub fork.
+
+First fork the `apple/swift` [repository](https://github.com/apple/swift.git),
+using the "Fork" button in the web UI, near the top-right. This will create a
+repository `username/swift` for your GitHub username. Next, add it as a remote:
+```sh
+# Using 'my-remote' as a placeholder name.
+
+# If you set up SSH in step 2
+git remote add my-remote git@github.com:username/swift.git
+
+# If you used HTTPS in step 2
+git remote add my-remote https://github.com/username/swift.git
+```
+Finally, create a new branch.
+```sh
+# Using 'my-branch' as a placeholder name
+git checkout my-branch
+git push --set-upstream my-remote my-branch
+```
+
+### First time Xcode setup
+
+If you used `--xcode` earlier, you will see an Xcode project generated under
+`../build/Xcode-RelWithDebInfoAssert/swift-macosx-x86_64`. When you open the
+project, Xcode might helpfully suggest "Automatically Create Schemes". Most of
```

Comment:
If desired, we could make `build-script` update the project do disable this feature for the project.

Reply:
PRs welcome! 😛

Reply:
I took a quick look, I don't see any obvious place to perform "post cmake generate actions". Is there such a place? Fwiw, the fix is to run:

```sh
plutil -replace IDEWorkspaceSharedSettings_AutocreateContextsIfNeeded -bool false \
    <Project>.xcodeproj/project.xcworkspace/xcshareddata/WorkspaceSettings.xcsettings
```

Reply:
I don't know; I'm not really familiar with this part of the code.